### PR TITLE
Add SetAssert

### PR DESCRIPTION
**Pull request by moaxcp**
This is to add SetAssert for assertions with vavr sets.

### CODE REVIEW

File: src/main/java/org/assertj/vavr/api/AbstractSetAssert.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import org.assertj.core.api.EnumerableAssert;
+
+import java.util.Comparator;
+
+abstract class AbstractSetAssert<SELF extends AbstractSetAssert<SELF, ACTUAL, ELEMENT, ELEMENT_ASSERT>,
+        ACTUAL extends Set<? extends ELEMENT>,
+        ELEMENT,
+        ELEMENT_ASSERT extends AbstractAssert<ELEMENT_ASSERT, ELEMENT>>
+        extends AbstractTraversableAssert<SELF, ACTUAL, ELEMENT, ELEMENT_ASSERT>
+        implements EnumerableAssert<SELF, ELEMENT> {
```
Comment (Opalo) [inline src/main/java/org/assertj/vavr/api/AbstractSetAssert.java:14]: I guess it cannot be implemented in a different way but it's very difficult to read.

File: src/main/java/org/assertj/vavr/api/VavrAssertions.java [hunk 2]
```diff
@@ -13,10 +13,7 @@
 package org.assertj.vavr.api;
 
 import io.vavr.Lazy;
-import io.vavr.collection.List;
-import io.vavr.collection.Map;
-import io.vavr.collection.Multimap;
-import io.vavr.collection.Seq;
+import io.vavr.collection.*;
```
Comment (Opalo) [inline src/main/java/org/assertj/vavr/api/VavrAssertions.java:16]: Does it wrap imports if `length(package.name)>4`? Is there any ruleset defined?

File: src/main/java/org/assertj/vavr/api/VavrAssumptions.java [hunk 3]
```diff
@@ -126,6 +127,19 @@ public static Object intercept(@This AbstractVavrAssert<?, ?> assertion, @SuperC
         return asAssumption(OptionAssert.class, Option.class, actual);
     }
 
+    /**
+     * Creates a new instance of <code>{@link SetAssert}</code> assumption.
```
Comment (Opalo) [inline src/main/java/org/assertj/vavr/api/VavrAssumptions.java:131]: `assumption` or `assertion`?

File: src/main/java/org/assertj/vavr/api/VavrAssumptions.java [hunk 4]
```diff
@@ -126,6 +127,19 @@ public static Object intercept(@This AbstractVavrAssert<?, ?> assertion, @SuperC
...
+     *
+     * @param <ELEMENT> type of elements contained in the {@link Set}.
+     * @param actual  the actual value.
+     * @return the created assumption for assertion object.
```
Comment (Opalo) [inline src/main/java/org/assertj/vavr/api/VavrAssumptions.java:135]: Ok, I see -  `Assumption`.

File: src/test/java/org/assertj/vavr/api/SetAssert_allSatisfy_Test.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        });
+    }
+
+    @Test
+    void should_fail_if_consumer_is_null() {
+        assertThatThrownBy(
+                () -> assertThat(HashSet.empty()).allSatisfy(null)
+        )
+                .isInstanceOf(NullPointerException.class)
+                .hasMessage("The Consumer<T> expressing the assertions requirements must not be null");
```
Comment (Opalo) [inline src/test/java/org/assertj/vavr/api/SetAssert_allSatisfy_Test.java:38]: `.` at the end?

File: src/main/java/org/assertj/vavr/api/AbstractSetAssert.java (hunk 1 above)
Comment (moaxcp) [reply src/main/java/org/assertj/vavr/api/AbstractSetAssert.java:14]: I agree!

File: src/main/java/org/assertj/vavr/api/VavrAssertions.java (hunk 2 above)
Comment (moaxcp) [reply src/main/java/org/assertj/vavr/api/VavrAssertions.java:16]: This may be due to my own settings. I can change it back if needed.

**Pull request merged**